test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -10556,6 +10556,393 @@ BOOST_AUTO_TEST_CASE(snark)
+		function f() public returns (fixed) {
+			return 3.5 / -1.25;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	BOOST_CHECK(callContractFunction("f()") == encodeArgs(createFixed(-28, 10, 19)));
+}
+
+
+BOOST_AUTO_TEST_CASE(ufixed_variable_division)
+{
+	char const* sourceCode = R"(
+	contract test {
+		ufixed a = 3.5;
+		ufixed b = 1.25;
+
+		function f() public returns (ufixed) {
+			return a / b;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	BOOST_CHECK(callContractFunction("f()") == encodeArgs(createFixed(28, 10, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_variable_division)
+{
+	char const* sourceCode = R"(
+	contract test {
+		fixed a = 3.5;
+		fixed b = -1.25;
+
+		function f() public returns (fixed) {
+			return a / b;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	BOOST_CHECK(callContractFunction("f()") == encodeArgs(createFixed(-28, 10, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(ufixed_comparison)
+{
+	char const* sourceCode = R"(
+	contract test {
+		ufixed a = 3.5;
+		ufixed b = 1.25;
+
+		function f() public returns (bool) {
+			return a > b;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	BOOST_CHECK(callContractFunction("f()") == encodeArgs(true));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_comparison)
+{
+	char const* sourceCode = R"(
+	contract test {
+		fixed a = -3.5;
+		fixed b = -1.25;
+
+		function f() public returns (bool) {
+			return a > b;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	BOOST_CHECK(callContractFunction("f()") == encodeArgs(false));
+}
+
+BOOST_AUTO_TEST_CASE(ufixed_constant_exponentiation)
+{
+	char const* sourceCode = R"(
+    contract test {
+        function f() public returns (ufixed) {
+            return 3.5 ** 3;
+        }
+    }
+    )";
+	compileAndRun(sourceCode, 0, "test");
+	BOOST_CHECK(callContractFunction("f()") == encodeArgs(createFixed(42875, 1000, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_constant_exponentiation)
+{
+	char const* sourceCode = R"(
+    contract test {
+        function f() public returns (fixed) {
+            return -3.5 ** 3;
+        }
+    }
+    )";
+	compileAndRun(sourceCode, 0, "test");
+	BOOST_CHECK(callContractFunction("f()") == encodeArgs(createFixed(-42875, 1000, 19)));
+}
+
+
+BOOST_AUTO_TEST_CASE(ufixed_variable_exponentiation)
+{
+	char const* sourceCode = R"(
+    contract test {
+		ufixed a = 3.5;
+		ufixed b = 3;
+
+        function f() public returns (ufixed) {
+            return a ** b;
+        }
+    }
+    )";
+	compileAndRun(sourceCode, 0, "test");
+	BOOST_CHECK(callContractFunction("f()") == encodeArgs(createFixed(42875, 1000, 19)));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_variable_exponentiation)
+{
+	char const* sourceCode = R"(
+    contract test {
+		fixed a = -3.5;
+		fixed b = 3;
+
+        function f() public returns (fixed) {
+            return a ** b;
+        }
+    }
+    )";
+	compileAndRun(sourceCode, 0, "test");
+	BOOST_CHECK(callContractFunction("f()") == encodeArgs(createFixed(-42875, 1000, 19)));
+}
+
```

Comment:
Please add tests with different fixed point types and tests that can create overflow